### PR TITLE
Archie/logger/custom logger

### PR DESCRIPTION
**Pull request by archiepeterjarvis**
# Implement custom logger

[Jira Ticket](https://yetitoolsmartsoftware.atlassian.net/jira/software/projects/YTSS/boards/1?assignee=5f8db15e6bc6340068c0bfc3&selectedIssue=YTSS-1028)
- [x] Ready for review

## Checklist
- [x] I have completed a self review
- [ ] I have updated the [spreadsheet](https://docs.google.com/spreadsheets/d/1277nPOKtBFOk_kCo870_zA460-qlw_kGnYlLPwtnadE/edit#gid=0)
- [ ] I have updated the jira ticket
- [ ] I have added the relevant labels to this PR
- [ ] I have updated documentation (if applicable)
- [ ] I have run the unit tests suite and they pass
- [ ] I have updated the [change log](https://docs.google.com/document/d/1t3s6lQuUcug1Fj8E_i2gBJNvXN0-funwBArG2hsia6k/edit)

## Description
- Removed all log() functions
- Updated all old log() function to use Logger.info()

## Notes
Setting the level of the logger also affects the Kivy logger, so if it's set to DEBUG we will also get Kivy's debug logs, which is a bit annoying.

### Visual Test
- [x] Not applicable
- [ ] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Function Test
- [ ] Not applicable
- [x] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Unit Tests
- [ ] Not applicable
- [ ] Completed

## Screenshots (if applicable)

### CODE REVIEW

File: src/asmcnc/comms/logging_system/logging_system.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    @staticmethod
+    def __get_console_handler(level):
+        """
+        Get a console handler for the logger.
+
+        :return:  A console handler for the logger.
+        """
+        console_handler = logging.StreamHandler(stream=sys.stdout)
+        console_handler.setLevel(level)
+        console_handler.setFormatter(
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/comms/logging_system/logging_system.py:100]: Yep good idea

File: src/asmcnc/production/lower_beam_qc_jig/widget_lower_beam_qc_xy_move.py [hunk 2]
```diff
@@ -134,11 +134,11 @@
                 background_color: hex('#F4433600')
                 always_release: True
                 on_release:
-                    print('release')
+                    Logger.info('release')
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/production/lower_beam_qc_jig/widget_lower_beam_qc_xy_move.py:137]: Will just remove this log

File: src/asmcnc/apps/systemTools_app/screens/calibration/screen_calibration_test.py [hunk 3]
```diff
@@ -1709,7 +1705,7 @@ def insert_stage_into_database(self, stage_id):
         try:
             self.calibration_db.insert_final_test_stage(self.sn_for_db, stage_id)
         except:
-            log("Could not insert final test stage into DB!!")
-            print(traceback.format_exc())
+            Logger.info("Could not insert final test stage into DB!!")
+            Logger.info(traceback.format_exc())
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/apps/systemTools_app/screens/calibration/screen_calibration_test.py:1709]: [13:45:27 07-03-2024] - [INFO] [__main__] Traceback (most recent call last):
  File "/Users/archiejarvis/PycharmProjects/easycut/easycut-smartbench/src/asmcnc/comms/logging_system/logging_system.py", line 127, in <module>
    int("a")
ValueError: invalid literal for int() with base 10: 'a'

works just fine

File: src/asmcnc/comms/logging_system/logging_system.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
+import inspect
+import logging
+import os
+import sys
+
+from colorlog import ColoredFormatter
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/comms/logging_system/logging_system.py:6]: It wouldn't be coloured in the logs.txt anyway, so I think this is just a dev thing.

**Review by Masc999 — APPROVED** [review]

**Pull request merged**